### PR TITLE
Add embedding op

### PR DESCRIPTION
**Pull request by wconstab**
#65576

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp [hunk 1]
```diff
@@ -51,6 +51,16 @@ namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_embedding(const at::Tensor & weight, const at::Tensor & indices, int64_t padding_idx, bool scale_grad_by_freq, bool sparse){
+  // Based on aten/src/ATen/native/Embedding.cpp::embedding.
```
Comment (alanwaketan) [inline lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:55]: Looks right to me. @bdhirsh mind take another look?

**Review by alanwaketan — APPROVED** [review]
LGTM.

File: lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp [hunk 2]
```diff
@@ -51,6 +51,16 @@ namespace torch_lazy_tensors {
...
+  std::vector<int64_t> out_sizes = indices.sizes().vec();
+  out_sizes.emplace_back(weight.size(1));
```
Comment (bdhirsh) [inline lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:57]: I think this is technically different from https://github.com/pytorch/pytorch/blob/a510f4139b2de1f5baed13bd50a035e39fd2a78b/aten/src/ATen/native/Embedding.cpp#L29, which uses `ArrayRef<T>::slice`.

`ArrayRef<T>`::slice looks like it grabs every element starting at weight.size(1) through the end (docs [here](https://github.com/pytorch/pytorch/blob/a510f4139b2de1f5baed13bd50a035e39fd2a78b/c10/util/ArrayRef.h#L207)).

Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:57]: Yea, it makes sense.

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:57]: but there is a torch.check that weight is a 2-D tensor,  right?  so it should be equiv?

Comment (bdhirsh) [reply lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:57]: oh yep, good point.

**Review by bdhirsh — APPROVED** [review]

**Pull request merged**